assets/app/scripts/directives/overviewDeployment.js

```diff
@@ -27,27 +28,39 @@ angular.module('openshiftConsole')
           $scope.desiredReplicas = null;
         });
 
-        // Debounce scaling so multiple clicks within 500 milliseconds only
-        // result in one request.
+        $scope.canScale = function() {
+          if (!$scope.deploymentConfigId || $scope.deploymentConfigMissing) {
+            // Scale as replication controller.
+            return true;
+          }
+
+          if (!$scope.deploymentConfig) {
+            // Wait for the deployment config.
+            return false;
+          }
+
+          // Only allow scaling of the latest deployment.
+          return deploymentIsLatestFilter($scope.rc, $scope.deploymentConfig);
```

Comment:
We talked about this on IRC, but I wanted to bring it here: No matter which RC you let users scale, the console will be making a call to scale the deploymentConfig, not an RC- and there will be a an indeterminate period of time where `replicationController.spec.replicas` changed via the UI widget will not be equal to `deploymentConfig.spec.replicas` (the latter being the target of the scale operation.)

Something to keep in mind if the UI re-syncs with the underlying RC.